core/src/main/java/hudson/WebAppMain.java

```diff
@@ -267,19 +268,27 @@ public void joinInit() throws InterruptedException {
      * Creates the plugin manager to use, if customized.
      * @return The plugin manager to use or {@code null} to use the default one.
      */
-    private PluginManager createPluginManager(ServletContext context, File rootDir) {
+    private @CheckForNull PluginManager createPluginManager(ServletContext context, File rootDir) {
         final String pmClassName = context.getInitParameter(PluginManager.class.getName() + ".className");
-        if (pmClassName == null) {
+        if (StringUtils.isBlank(pmClassName)) {
             return null;
         }
         try {
             final Class<? extends PluginManager> klass = Class.forName(pmClassName).asSubclass(PluginManager.class);
             final Constructor <? extends PluginManager> constructor = klass.getConstructor(ServletContext.class, File.class);
             return constructor.newInstance(context, rootDir);
+        } catch(NullPointerException e) {
+            // Class.forName and Class.getConstructor are supposed to never return null, but just in case we introduce
+            // this specific catch to avoid polluting the logs with NPEs.
+            LOGGER.log(WARNING, String.format("Unable to instantiate custom plugin manager %s. Using default.", pmClassName));
+        } catch(ClassCastException e) {
+            LOGGER.log(WARNING, String.format("Provided class %s does not extend PluginManager. Using default.", pmClassName));
+        } catch(NoSuchMethodException e) {
+            LOGGER.log(WARNING, String.format("Provided custom plugin manager %s does not provided (ServletContext, File) constructor. Using default.", pmClassName));
```

Comment:
NIT: I would still expose the original stacktrace. Just in case there is any hidden binary issue for custom cores